backend.py

```diff
@@ -49,12 +49,10 @@ def walk(self, distance, state, direction=None):
             direction = self.direction
 
         # Robot can go backwards - then his distance is -1.
-        # In this case we want to rotate him, make him walk 1 step and rotate back.
-        # He still can move the other robots on the way.
+        # In this case he walks 1 step in the direction opposite to his.
+        # He can still move the other robots on the way.
         if distance < 0:
-            self.rotate(Rotation.U_TURN)
-            self.walk((-distance), state)
-            self.rotate(Rotation.U_TURN)
+            self.walk((-distance), state, self.direction.get_new_direction(Rotation.U_TURN))
```

Comment:
Můžu poprosit o vysvětlení? Jak to bylo rozepsané do těch třech řádků tak to chápu jak se postupně hýbe, ale tohle ne. 
Nejdřív couvne směrem který má? Mate mě proč je tam Rotation.U_TURN, nebude potom stát obráceně? Děkuji.

Reply:
Robot se tu netočí. Vybírá se jen směr pohybu.
Třeba místo „o `-1` políčko na sever“ se pohne „o `1` políčko na jih“.

Reply:
Musíme zadat robotovi jako parametr směr "o 180" stupňů otočen, než ten, ve kterém dostal příkaz jít. My mu řeknem: `jdi -1 krok na sever`, což v jeho jazyku znamená `jdi +1 krok na jih`. Jeho vlastní směr zde nehraje roli (mrkni na nový commit, který to snad řeší), otačíme ho podle směru, který se po něm požaduje. 
Takže abych šla robotem dozadu (v libovolném směru), musím jít kladný počet kroků dopředu - ale ve směru opačném k zadanému. Směr samotného robota nikde neměním.

Reply:
A když mu řeknu jdi o `-1` na západ tak se posune na západ, ale koukat jakože otočený bude stále stejně na sever?

Reply:
Ano, presne tak. To (Rotation.U_TURN) se nevztahuje k jeho vlastnimu otoceni, ale smeru, ve kterem ma jit.